Automattic/newspack-newsletters
# feat: posts inserter enhancements

### All Submissions:






### Changes proposed in this Pull Request:



Two changes:

1. Adds a new "Display author" option that displays bylines if enabled. Closes #332.
2. Adds Tags, Excluded Tags, and Excluded Categories options in an "advanced filters" panel. Closes #234.

### How to test the changes in this Pull Request:

#### Testing "Display author" option

1. Check out this branch.
2. In a new or existing Newsletter draft containing a Posts Inserter block, click the block and observe a new "Display author" option which is DISABLED by default.
3. Enable the new option. Confirm that the posts shown in the Posts Inserter preview now appear with a byline.
4. Send the newsletter or send a test email. Confirm that the posts shown in the sent email also contain the bylines.

#### Testing new filter options

1. In a new or existing Newsletter draft containing a Posts Inserter block, observe a new "show advanced filters" button underneath the Category field.
2. Click the "show advanced filters" button. Observe that three new fields appear: Tags, Excluded Categories, and Excluded Tags.
3. Type post names into the fields and confirm behavior:
    - If selecting one or more tag, confirm that posts are only shown if they own one of the selected tags.
    - If selecting one or more Excluded Categories, confirm that posts are only shown if they do NOT own one of the selected Excluded Categories.
    - If selecting one or more Excluded Tags, confirm that posts are only shown if they do NOT own one of the selected Excluded Tags.
    - Also confirm that the existing query settings (Display Specific Posts, Order, and Number of Items) still work as expected.
4. Send the newsletter or a test email and confirm that the sent email has the same posts as shown in the Posts Inserter block editor.

Note that it's possible to create an impossible query by selecting the same category in both the Categories and Excluded Categories fields—this is also true in the Homepage Posts block, so I felt it was at least consistent.

Sidenote: This is another instance where https://github.com/Automattic/newspack-plugin/pull/674 could be useful, although I don't know if it would cause trouble now that this plugin is available via dotorg.

### Other information:
